src/py/mat3ra/made/tools/build/interface.py

```diff
@@ -1,77 +1,199 @@
-        layer_parameters=SlabParameters(miller_indices=(0, 0, 1), thickness=1),
-        distance_z=3.0,
-        use_conventional_cell=True,
-        zsl_parameters=ZSLParameters(),
+        substrate_configuration: SlabConfiguration,
+        film_configuration: SlabConfiguration,
+        termination_pair: TerminationPair,
+        distance_z: float = 3.0,
+        vacuum: float = 5.0,
     ):
-        self.SubstrateParameters = substrate_parameters
-        self.LayerParameters = layer_parameters
+        super().__init__()
+        self.substrate_configuration = substrate_configuration
+        self.film_configuration = film_configuration
+        self.termination_pair = termination_pair
+        self.vacuum: float = vacuum
+        self.distance_z: float = distance_z
+
+    @property
+    def bulk(self):
+        # Return the bulk structure of the interface configuration, with no vacuum
+        return self.get_material(vacuum=0)
+
+    @property
+    def miller_indices(self):
+        return self.__miller_indices
+
+    def get_material(self, vacuum=None, scale_film_to_fit: bool = False):
+        vacuum = self.vacuum if vacuum is None else vacuum
+        substrate_slab = self.substrate_configuration.get_material(termination=self.termination_pair[1])
+        film_slab = self.film_configuration.get_material(termination=self.termination_pair[0])
+
+        substrate_slab_ase = to_ase(substrate_slab)
+        film_slab_ase = to_ase(film_slab)
+
+        # Reduce the cell to Niggli form if desired
+        niggli_reduce(substrate_slab_ase)
+        niggli_reduce(film_slab_ase)
+
+        if scale_film_to_fit:
+            film_slab_ase.set_cell(substrate_slab_ase.cell, scale_atoms=True)
+            film_slab_ase.wrap()
+
+        # Calculate z-shift based on the new positions after scaling
+        max_z_substrate = max(substrate_slab_ase.positions[:, 2])
+        min_z_film = min(film_slab_ase.positions[:, 2])
+        shift_z = max_z_substrate - min_z_film + self.distance_z
+
+        film_slab_ase.translate([0, 0, shift_z])
+
+        interface_ase = substrate_slab_ase + film_slab_ase
+
+        # Adjust the cell height to include the vacuum space above the film
+        cell_c_with_vacuum = max(interface_ase.positions[:, 2]) + vacuum
+        interface_ase.cell[2, 2] = cell_c_with_vacuum
+
+        material_dict = from_ase(interface_ase)
+        return Material(material_dict)
+
+    @property
+    def interface_properties(self):
+        return {
+            "substrate_configuration": self.substrate_configuration,
+            "film_configuration": self.film_configuration,
+            "termination_pair": self.termination_pair,
+            "distance_z": self.distance_z,
+            "vacuum": self.vacuum,
+        }
 
-        self.distance_z = distance_z
-        self.use_conventional_cell = use_conventional_cell
 
-        self.ZSLParameters = zsl_parameters
+class ScaleInterfaceBuilder:
```

Comment:
1. SimpleInterfaceBuilder or BasicInterfaceBuilder
2. Should contain function `build` that does the logic on lines 63-88 above
3. InterfaceConfiguration should use the BaseInterfaceBuilder.build in get_material